step-templates/roundhouse-database-migration.json

```diff
@@ -22,7 +22,7 @@
[string]::IsNullOrEmpty($roundhouseUpFolder))\n{\n    # Add up folder\n    $roundhouseSwitches += \"--up=$roundhouseUpFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseAlterDatabaseFolder))\n{\n    $roundhouseSwitches += \"--alterdatabasefolder=$roundhouseAlterDatabaseFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseRunBeforeUpFolder))\n{\n    $roundhouseSwitches += \"--runbeforeupfolder=$roundhouseRunBeforeUpFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseFunctionsFolder))\n{\n    $roundhouseSwitches += \"--functionsfolder=$roundhouseFunctionsFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseViewsFolder))\n{\n    $roundhouseSwitches += \"--viewsfolder=$roundhouseViewsFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseSprocsFolder))\n{\n    $roundhouseSwitches += \"--sprocsfolder=$roundhouseSprocsFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseIndexFolder))\n{\n    $roundhouseSwitches += \"--indexesfolder=$roundhouseIndexFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseRunAfterAnyTimeFolder))\n{\n    $roundhouseSwitches += \"--runAfterOtherAnyTimeScriptsFolder=$roundhouseRunAfterAnyTimeFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhousePermissionsFolder))\n{\n    $roundhouseSwitches += \"--permissionsfolder=$roundhousePermissionsFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseTriggerFolder))\n{\n    $roundhouseSwitches += \"--triggersfolder=$roundhouseTriggerFolder\"\n}\n\nif ([System.Boolean]::Parse($roundhouseDryRun))\n{\n    $roundhouseSwitches += \"--dryrun\"\n}\n\nif ([System.Boolean]::Parse($roundhouseRecordOutput))\n{\n    $roundhouseSwitches += \"--outputpath=$roundhouseOutputPath\"\n}\n\n# Add transaction switch\n$roundhouseSwitches += \"--withtransaction=$($roundhouseWithTransaction.ToLower())\"\n\n# Check for version\nif (![string]::IsNullOrEmpty($roundhouseVersion))\n{\n    # Add version\n    $roundhouseSwitches += \"--version=$roundhouseVersion\"\n}\n\n# Display what's going to be run\nif (![string]::IsNullOrWhitespace($roundhouseUserPassword))\n{\n\tWrite-Host \"Executing $($roundhouseExecutable.FullName) with $($roundhouseSwitches.Replace($roundhouseUserPassword, \"****\"))\"\n}\nelse\n{\n\tWrite-Host \"Executing $($roundhouseExecutable.FullName) with $($roundhouseSwitches)\"\n}\n\n# Execute RoundhousE\nif ($roundhouseExecutable.FullName.EndsWith(\".dll\"))\n{\n\t& dotnet $roundhouseExecutable.FullName $roundhouseSwitches\n}\nelse\n{\n\t& $roundhouseExecutable.FullName $roundhouseSwitches\n}\n\n# If the output path was specified, attach artifacts\nif ([System.Boolean]::Parse($roundhouseRecordOutput))\n{    \n    # Zip up output folder content\n    Add-Type -Assembly 'System.IO.Compression.FileSystem'\n    \n    $zipFile = \"$($OctopusParameters[\"Octopus.Action.Package[RoundhousEPackage].ExtractedPath\"])\\output.zip\"\n    \n\t[System.IO.Compression.ZipFile]::CreateFromDirectory($roundhouseOutputPath, $zipFile)\n    New-OctopusArtifact -Path \"$zipFile\" -Name \"output.zip\"\n}\n"
```

Comment:
Suggestions on code change:

1. Spelling:
Use of `$roundhouseAuthenticaitonMethod` should be `$roundhouseAuthenticationMethod` 

2. **Optional:** Use Path's Combine method(avoids hard coded use of either `\` or `/` 

e.g.
```ps
$roundhouseOutputPath = "$($OctopusParameters["Octopus.Action.Package[RoundhousEPackage].ExtractedPath"])/output"
```
becomes:

```ps
$roundhouseOutputPath = [System.IO.Path]::Combine($OctopusParameters["Octopus.Action.Package[RoundhousEPackage].ExtractedPath"], "output")
```

3. **Optional:** Reduce duplicated search for rh.dll

```ps
# Set Executable depending on OS
if ($IsWindows)
{
  # Look for older .exe
  $roundhouseExecutable = Get-ChildItem -Path $PSSCriptRoot -Recurse | Where-Object {$_.Name -eq "rh.exe"}

  if ($null -eq $roundhouseExecutable)
  {
      # Look for dotnet core dll
      $roundhouseExecutable = Get-ChildItem -Path $PSSCriptRoot -Recurse | Where-Object {$_.Name -eq "rh.dll"}
  }
}

if ($IsLinux)
{
	# Look for just rh.dll
    $roundhouseExecutable = Get-ChildItem -Path $PSSCriptRoot -Recurse | Where-Object {$_.Name -eq "rh.dll"}
}

if ([string]::IsNullOrWhitespace($roundhouseExecutable))
{
    # Couldn't find RoundhousE
    Write-Error "Couldn't find the RoundhousE executable!"
}
```
becomes:

```ps
# Find executable (dependent on OS)
if ($IsWindows)
{
  # Look for older .exe
  $roundhouseExecutable = Get-ChildItem -Path $PSSCriptRoot -Recurse | Where-Object {$_.Name -eq "rh.exe"}
}

# If not found yet, look for rh.dll
if([string]::IsNullOrWhitespace($roundhouseExecutable)) 
{    
    $roundhouseExecutable = Get-ChildItem -Path $PSSCriptRoot -Recurse | Where-Object {$_.Name -eq "rh.dll"}
}

if ([string]::IsNullOrWhitespace($roundhouseExecutable))
{
    # Couldn't find RoundhousE
    Write-Error "Couldn't find the RoundhousE executable!"
}
```
4. I don't know if this is an issue or not:

The following code has been removed and isn't in the new version
```ps
# Check to see if username was specified
if ([string]::IsNullOrEmpty($roundhouseUsername))
{
    $roundhouseUserInfo = "Trusted_Connection=True;"
}
```